demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py

```diff
@@ -22,46 +22,65 @@ class MaskRCNN(IEModel):
+               required_output_keys_segmentoly.issubset(self.net.outputs)
 
-        self.n, self.c, self.h, self.w = self.inputs_info['im_data'].input_data.shape
+        input_name = 'im_data' if self.segmentoly_type() else 'image'
+        self.n, self.c, self.h, self.w = self.inputs_info[input_name].input_data.shape
+
+    def segmentoly_type(self):
+        return 'image' not in set(self.inputs_info.keys())
 
     def get_allowed_inputs_len(self):
-        return (2, )
+        return (1, 2)
 
     def get_allowed_outputs_len(self):
-        return (4, 5)
+        return (3, 4, 5)
 
     def _preprocess(self, frame):
         image_height, image_width = frame.shape[:2]
         scale = min(self.h / image_height, self.w / image_width)
         processed_image = cv2.resize(frame, None, fx=scale, fy=scale)
         processed_image = processed_image.astype('float32').transpose(2, 0, 1)
-        im_info=np.array([processed_image.shape[1], processed_image.shape[2], 1.0], dtype='float32')
+        im_info = np.array([processed_image.shape[1], processed_image.shape[2], 1.0], dtype='float32')
+        if not self.segmentoly_type():
+            im_info = None
         meta=dict(original_size=frame.shape[:2],
                   processed_size=processed_image.shape[1:3])
         return processed_image, im_info, meta
 
     def forward(self, im_data, im_info):
+        input_name = 'im_data' if self.segmentoly_type() else 'image'
         if (self.h - im_data.shape[1] < 0) or (self.w - im_data.shape[2] < 0):
             raise ValueError('Input image should have the resolution of {}x{} or less, '
                              'got {}x{}.'.format(self.w, self.h, im_data.shape[2], im_data.shape[1]))
         im_data = np.pad(im_data, ((0, 0),
                                    (0, self.h - im_data.shape[1]),
                                    (0, self.w - im_data.shape[2])),
                          mode='constant', constant_values=0).reshape(1, self.c, self.h, self.w)
-        im_info = im_info[None, ]
-        output = self.net.infer(dict(im_data=im_data, im_info=im_info))
-
-        classes = output['classes']
-        valid_detections_mask = classes > 0
-        classes = classes[valid_detections_mask]
-        boxes = output['boxes'][valid_detections_mask]
-        scores = output['scores'][valid_detections_mask]
-        masks = output['raw_masks'][valid_detections_mask]
+        feed_dict = {input_name: im_data}
+        if im_info is not None:
+            im_info = im_info.reshape(1, *im_info.shape)
+            feed_dict['im_info'] = im_info
+        output = self.net.infer(feed_dict)
+
+        if self.segmentoly_type():
+            valid_detections_mask = output['classes'] > 0
+            classes = output['classes'][valid_detections_mask]
+            boxes = output['boxes'][valid_detections_mask]
+            scores = output['scores'][valid_detections_mask]
+            masks = output['raw_masks'][valid_detections_mask]
```

Comment:
Reimplemented method `segmentoly_type()` with checking all inputs and outputs.

Reply:
It's not quite sufficient, because your `__init__` method still admits models with segmentoly inputs and non-segmentoly outputs, and vice versa. Such models will be treated as non-segmentoly, and either input or output processing will fail.

Reply:
Reimplemented.